mono/metadata/icall.c

```diff
@@ -774,61 +761,66 @@ ves_icall_System_Array_FastCopy (MonoArray *source, int source_idx, MonoArray* d
 			return FALSE;
 	}
 
+	gboolean pin = FALSE; // FIXME Making a gchandle is inefficient and breaks profiler tests.
+	gchandle_t source_gchandle = 0;
+	gchandle_t dest_gchandle = 0;
+
 	if (m_class_is_valuetype (dest_class)) {
-		element_size = mono_array_element_size (source->obj.vtable->klass);
-		source_addr = mono_array_addr_with_size_fast (source, element_size, source_idx);
+		gsize const element_size = mono_array_element_size (MONO_HANDLE_GETVAL (source, obj.vtable->klass));
```

Comment:
I think this block is another great place to use `MONO_ENTER_NO_SAFEPOINTS`.
And to delete `gboolean pin`.

Reply:
I have no other known options, so yeah. Maybe could be builtin to the wrapper to reduce TLS access.

Reply:
Would also be nice for icall wrapper to pass down the `ThreadInfo`.